src/test/java/test/EmployeeTest.java

```diff
@@ -0,0 +1,129 @@
+                        .employeeAge(66)
+                        .profileImage("")
+                        .build()},
+        };
+    }
+
+    @Test(dataProvider = "existingEmployees")
+    public void checkGetEmployeesByIds(EmployeeData employeeData) {
+        Employee employee = getEmployee(employeeData.getId());
+
+        SoftAssertions softly = new SoftAssertions();
+        softly.assertThat(employee.getData().getEmployeeName()).isEqualTo(employeeData.getEmployeeName());
+        softly.assertThat(employee.getData().getEmployeeSalary()).isEqualTo(employeeData.getEmployeeSalary());
+        softly.assertThat(employee.getData().getEmployeeAge()).isEqualTo(employeeData.getEmployeeAge());
+        softly.assertThat(employee.getData().getProfileImage()).isEqualTo(employeeData.getProfileImage());
+        softly.assertAll();
+    }
+
+    @Test
+    public void checkForEmployeeWithNotExistingId() {
+        int notExistingId = 25;
+        Employee employee = getEmployee(notExistingId);
+
+        SoftAssertions softly = new SoftAssertions();
+        softly.assertThat(employee.getData()).isNull();
+        softly.assertThat(employee.getStatus()).isEqualTo("failure");
+        softly.assertThat(employee.getMessage()).isEqualTo("Failure! There is no such employee.");
+        softly.assertAll();
+    }
+
+    @Test
+    public void testCreateNewEmployee() {
+        NewEmployeeData newEmployeeData = NewEmployeeData.builder()
+                .name("Adam")
+                .salary(5000)
+                .age(40)
+                .build();
+
+        NewEmployee employeeToBeCreated = postEmployee(newEmployeeData);
+
+        int newEmployeeId = employeeToBeCreated.getData().getId();
+
+        Employee createdEmployee = getEmployee(newEmployeeId);
+
+        assertThat(createdEmployee.getData()).as("check if new employee has not null data").isNotNull();
+
+        SoftAssertions softly = new SoftAssertions();
+        softly.assertThat(employeeToBeCreated.getMessage()).isEqualTo("Successfully! Record has been added.");
+        softly.assertThat(createdEmployee.getData().getId()).isEqualTo(employeeToBeCreated.getData().getId());
+        softly.assertThat(createdEmployee.getData().getEmployeeName()).isEqualTo(employeeToBeCreated.getData().getName());
+        softly.assertThat(createdEmployee.getData().getEmployeeAge()).isEqualTo(employeeToBeCreated.getData().getAge());
+        softly.assertThat(createdEmployee.getData().getEmployeeSalary()).isEqualTo(employeeToBeCreated.getData().getSalary());
+        softly.assertAll();
+    }
+
+    private Employee getEmployee(int id) {
+        Response employeeByIdResponse =
+                given()
+                        .spec(requestSpecification)
+                        .pathParams("id", id)
+                .when()
+                        .get("/employee/{id}");
+
+        assertThat(employeeByIdResponse.statusCode()).as("check response status code").isEqualTo(200);
```

Comment:
in case of status check you can use hamcrest assertion instead of assertj and return Employee right away